zio-config/src/main/scala/zio/config/ConfigDescriptor.scala

```diff
@@ -58,15 +60,17 @@ object ConfigDescriptor {
 
   final case class Source[A](path: String, propertyType: PropertyType[A]) extends ConfigDescriptor[A]
 
+  final case class Nested[A](config: ConfigDescriptor[A], path: String) extends ConfigDescriptor[A]
+
   final case class Describe[A](config: ConfigDescriptor[A], message: String) extends ConfigDescriptor[A]
 
-  final case class Default[A](configDescriptor: ConfigDescriptor[A], value: A) extends ConfigDescriptor[A]
+  final case class Default[A](config: ConfigDescriptor[A], value: A) extends ConfigDescriptor[A]
 
   final case class Optional[A](config: ConfigDescriptor[A]) extends ConfigDescriptor[Option[A]]
 
   final case class XmapEither[A, B](
     config: ConfigDescriptor[A],
-    f: A => Either[ReadError, B],
+    f: A => Either[ReadError[String, String], B],
```

Comment:
You could potentially push `K` and `V` into `ConfigDescriptor`, and make them covariant, but perhaps that's something we should look at later.